TextExtractor-20388: add SupportedLanguages and Troubleshooting sections

taras-janea opened the pull request.

taras-janea committed:
TextExtractor-20388: add SupportedLanguages and Troubleshooting sections

taras-janea commented:
cc: @jaimecbernardo

opbld33 commented:
Learn Build status updates of commit _[338306e](https://github.com/taras-janea/windows-dev-docs/commits/338306e714993b1749e5cfc72af9bc84f876045b)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[hub/powertoys/text-extractor.md](https://github.com/taras-janea/windows-dev-docs/blob/TextExtractor-20388/hub/powertoys/text-extractor.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/8f0bf4e1-986f-a415-e316-169a990c8a5d/202209281304345518-4065/BuildReport?accessString=563a7b8283a6fbf2217c6c7811a59c37f0fc85fd2c0db93b378e7069591f94a4).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/8f0bf4e1-986f-a415-e316-169a990c8a5d?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

jaimecbernardo commented on hub/powertoys/text-extractor.md:
```diff
@@ -31,3 +36,60 @@ From the Settings menu, the following options can be configured:
+
+## Troubleshooting
+
+* "No Possible OCR languages are installed." message </br>
```
Please use `-` instead of `*` as in other pages.

jaimecbernardo commented on hub/powertoys/text-extractor.md:
```diff
@@ -31,3 +36,60 @@ From the Settings menu, the following options can be configured:
+
+## Supported languages
+
+Text Extractor can recognize only languages with OCR pack available.</br>
```
Text Extractor can only recognize languages that have the OCR pack installed.

jaimecbernardo commented on hub/powertoys/text-extractor.md:
```diff
@@ -31,3 +36,60 @@ From the Settings menu, the following options can be configured:
+## Supported languages
+
+Text Extractor can recognize only languages with OCR pack available.</br>
+The list can be obtained via PowerShell by running next commands:
```
"next commands" -> "following commands"

jaimecbernardo commented on hub/powertoys/text-extractor.md:
```diff
@@ -31,3 +36,60 @@ From the Settings menu, the following options can be configured:
+
+### OCR packs installation
+
+If language is supported, then OCR language pack is installed along with language via <kbd>Language</kbd>-><kbd>Preferred languages</kbd>-><kbd>Add language menu</kbd> by default.</br>
```
"For supported languages, the OCR language pack can be installed along with the language via"
And remove the "by default."

jaimecbernardo commented on hub/powertoys/text-extractor.md:
```diff
@@ -31,3 +36,60 @@ From the Settings menu, the following options can be configured:
+
+If language is supported, then OCR language pack is installed along with language via <kbd>Language</kbd>-><kbd>Preferred languages</kbd>-><kbd>Add language menu</kbd> by default.</br>
+
+It's possible to query all supported languages via PowerShell run as Administrator:
```
"It's possible to get a list of all supported languages by running this command in PowerShell in Administrator mode:"

jaimecbernardo commented on hub/powertoys/text-extractor.md:
````diff
@@ -31,3 +36,60 @@ From the Settings menu, the following options can be configured:
+....
+```
+
+If language is not available in the output, then it's not supported by OCR.</br>
````
"If a language"

jaimecbernardo commented on hub/powertoys/text-extractor.md:
```diff
@@ -31,3 +36,60 @@ From the Settings menu, the following options can be configured:
+
+If language is not available in the output, then it's not supported by OCR.</br>
+If the state is <kbd>NotPresent</kbd>, then it's possible to install OCR pack for the language.</br>
+For example, the next commands install OCR pack for <kbd>en-US</kbd> in PowerShell run as Administrator:
```
"install OCR pack" -> "install the OCR pack"

jaimecbernardo reviewed (changes requested):
I think this needs some language changes.
Instead of using <br> blocks, please use a blank line between sentences as it's done in other documentation pages.
Other than that, it LGTM!

jaimecbernardo commented:
Please add @crutkas to reviewers as well.

taras-janea committed:
TextExtractor-20388: fix code review comments from Jaime

opbld33 commented:
Learn Build status updates of commit _[af1dcbc](https://github.com/taras-janea/windows-dev-docs/commits/af1dcbc14c00c0716ba8a28be88beee247913ef0)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[hub/powertoys/text-extractor.md](https://github.com/taras-janea/windows-dev-docs/blob/TextExtractor-20388/hub/powertoys/text-extractor.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/8f0bf4e1-986f-a415-e316-169a990c8a5d/202209281419340256-4065/BuildReport?accessString=a30fba63463039019af1f954d9fd91bc4d7c4765d2a61bae5a42dc306c4df8ab).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/8f0bf4e1-986f-a415-e316-169a990c8a5d?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

opbld30 commented:
Learn Build status updates of commit _[66b1a7a](https://github.com/taras-janea/windows-dev-docs/commits/66b1a7aec171bb92b755dba7a33bb2216860ecc2)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[hub/powertoys/text-extractor.md](https://github.com/taras-janea/windows-dev-docs/blob/TextExtractor-20388/hub/powertoys/text-extractor.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/8f0bf4e1-986f-a415-e316-169a990c8a5d/202209281550564832-4065/BuildReport?accessString=446efb5b17a87c60a2549aac4a4df2cb3e761ba5cdd0a25a2abe2c15d3090b90).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/8f0bf4e1-986f-a415-e316-169a990c8a5d?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

jaimecbernardo commented on hub/powertoys/text-extractor.md:
````diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
+PS C:\Windows\system32> $Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
+PS C:\Windows\system32> $Capability | Add-WindowsCapability -Online
+```
+and these remove:
````
These commands remove it:

jaimecbernardo commented on hub/powertoys/text-extractor.md:
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
+
+- "No Possible OCR languages are installed." message
+
+Message is shown, when there are no available languages for recognition.
```
"This message is shown when there are no available languages for recognition."

jaimecbernardo commented on hub/powertoys/text-extractor.md:
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
+
+Message is shown, when there are no available languages for recognition.
+
+If OCR pack is supported, installed, but still not available and your system drive <kbd>X:</kbd> is different than <kbd>C:</kbd>, then copy <kbd>X:/Windows/OCR</kbd> folder to <kbd>C:/Windows/OCR</kbd> to fix the issue.
```
"If an OCR pack"

jaimecbernardo reviewed (approved):
Some nits, other than that LGTM! Thank you!

opbld30 commented:
Learn Build status updates of commit _[ebb8947](https://github.com/taras-janea/windows-dev-docs/commits/ebb8947539aa620096fa605a5e7f85ba6760c420)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[hub/powertoys/text-extractor.md](https://github.com/taras-janea/windows-dev-docs/blob/TextExtractor-20388/hub/powertoys/text-extractor.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/8f0bf4e1-986f-a415-e316-169a990c8a5d/202209281647325660-4065/BuildReport?accessString=0f888ce4493c82f80b9583233b44dd1d5e1bda558f392a18a68a0ab51792655b).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/8f0bf4e1-986f-a415-e316-169a990c8a5d?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

crutkas commented on hub/powertoys/text-extractor.md:
````diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
+PS C:\Windows\system32> $Capability | Remove-WindowsCapability -Online
+```
+
+## Troubleshooting
````
how does Install a language pack needs to be a section

taras-janea replied:
Have removed "### OCR packs installation" line. I guess that's what you meant, right?

crutkas replied:
no, if i don't have the OCR pack needed, we should say how they get it.  "Hey, i want german, ..."

taras-janea replied:
Hi Clint,

I've made changes according to your comments.
Could you please check whether it looks good to you?

Thank you,
Taras

mattwojo replied:
Looks good! Thank you Taras!

taras-janea committed:
TextExtractor-20388: removed section "### OCR packs installation" according to code review comments

opbld30 commented:
Learn Build status updates of commit _[5880de3](https://github.com/taras-janea/windows-dev-docs/commits/5880de38e385b82bb9202a4f5c868a3896beef8c)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[hub/powertoys/text-extractor.md](https://github.com/taras-janea/windows-dev-docs/blob/TextExtractor-20388/hub/powertoys/text-extractor.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/8f0bf4e1-986f-a415-e316-169a990c8a5d/202209291015495028-4065/BuildReport?accessString=75a503697074a9568e8043f7e82880a77396b72480ff1380b7712bb8beaacd2f).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/8f0bf4e1-986f-a415-e316-169a990c8a5d?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

taras-janea committed:
Revert "TextExtractor-20388: removed section "### OCR packs installation" according to code review comments"

This reverts commit 5880de38e385b82bb9202a4f5c868a3896beef8c.

opbld33 commented:
Learn Build status updates of commit _[aeec281](https://github.com/taras-janea/windows-dev-docs/commits/aeec2810c3a0847cadfe8df05262306686ac1557)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[hub/powertoys/text-extractor.md](https://github.com/taras-janea/windows-dev-docs/blob/TextExtractor-20388/hub/powertoys/text-extractor.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/8f0bf4e1-986f-a415-e316-169a990c8a5d/202209291918039141-4065/BuildReport?accessString=0adbc23fe8d0129947429947e4cf3ff19a3aa6315ca812ea09ffe636cb4cfeea).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/8f0bf4e1-986f-a415-e316-169a990c8a5d?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

crutkas commented on hub/powertoys/text-extractor.md:
````diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
+PS C:\Windows\system32> $Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
+PS C:\Windows\system32> $Capability | Add-WindowsCapability -Online
+```
+These commands remove it:
````
```suggestion
#### How to remove a language pack
```

crutkas commented on hub/powertoys/text-extractor.md:
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
+
+If a language is not available in the output, then it's not supported by OCR.
+
+If the state is <kbd>NotPresent</kbd>, then it's possible to install OCR pack for the language.
```
```suggestion
#### How to install a language pack
```

crutkas commented on hub/powertoys/text-extractor.md:
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
+
+For supported languages, the OCR language pack can be installed along with the language via <kbd>Language</kbd>-><kbd>Preferred languages</kbd>-><kbd>Add language menu</kbd>.
+
+It's possible to get a list of all supported languages by running this command in PowerShell in Administrator mode:
```
```suggestion
#### How to query for language packs
```

taras-janea committed:
TextExtractor-20388: make #SupportedLanguages section more structured

opbld31 commented:
Learn Build status updates of commit _[648b95d](https://github.com/taras-janea/windows-dev-docs/commits/648b95d3b0ce6949d3794a03f2b117c8dd7988c6)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[hub/powertoys/text-extractor.md](https://github.com/taras-janea/windows-dev-docs/blob/TextExtractor-20388/hub/powertoys/text-extractor.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/8f0bf4e1-986f-a415-e316-169a990c8a5d/202209301221039939-4065/BuildReport?accessString=51c48fbc8449c8ebb4b7fe3066ba6d62fbf874c889afd5b989622e7a6a385c45).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/8f0bf4e1-986f-a415-e316-169a990c8a5d?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

Jay-o-Way commented on hub/powertoys/text-extractor.md:
```diff
@@ -14,6 +14,12 @@ Text Extractor is a convenient way to copy text from anywhere on your screen. Th
+
 ## Adjust while trying to capture
 
 By holding <kbd>shift</kbd>, you will change from adjusting the capture region's size to moving the capture region. When you release <kbd>shift</kbd>, you will be able to resize again.
```
```suggestion
By holding <kbd>Shift</kbd>, you will change from adjusting the capture region's size to moving the capture region. When you release <kbd>Shift</kbd>, you will be able to resize again.
```
Caps

Jay-o-Way commented on hub/powertoys/text-extractor.md:
```diff
@@ -14,6 +14,12 @@ Text Extractor is a convenient way to copy text from anywhere on your screen. Th
 
+## How to deactivate
+
+Capture mode is deactivated immediately after text in the selected region is recognized and copied to the clipboard.
+
+You can exit capture mode by pressing <kbd>Esc</kbd> at any moment.
```
```suggestion
Capture mode is deactivated immediately after text in the selected region is recognized and copied to the clipboard. You can exit capture mode by pressing <kbd>Esc</kbd> at any moment.
```
Why new paragraph?

Jay-o-Way commented on hub/powertoys/text-extractor.md:
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
+
+This message is shown when there are no available languages for recognition.
+
+If an OCR pack is supported and installed, but still is not available and your system drive <kbd>X:</kbd> is different than <kbd>C:</kbd>, then copy <kbd>X:/Windows/OCR</kbd> folder to <kbd>C:/Windows/OCR</kbd> to fix the issue.
```
```suggestion
If an OCR pack is supported and installed, but still is not available and your system drive _X:_ is different than "C:", then copy `X:/Windows/OCR`  folder to `C:/Windows/OCR` to fix the issue.
```
These aren't keys on a keyboard.

Jay-o-Way commented on hub/powertoys/text-extractor.md:
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
+
+### How to remove an OCR language pack
+
+The following commands remove the OCR pack for <kbd>en-US</kbd>:
```
```suggestion
The following commands remove the OCR pack for "en-US":
```

Jay-o-Way commented on hub/powertoys/text-extractor.md:
```diff
@@ -31,3 +37,65 @@ From the Settings menu, the following options can be configured:
+
+### How to install an OCR language pack
+
+The following commands install the OCR pack for <kbd>en-US</kbd>:
```
```suggestion
The following commands install the OCR pack for "en-US":
```

Jay-o-Way reviewed (changes requested):
kbd tags & syntax

taras-janea committed:
TextExtractor-20388: fix code review comments from Jay-o-Way

opbld32 commented:
Learn Build status updates of commit _[45241eb](https://github.com/taras-janea/windows-dev-docs/commits/45241ebfd0f23260f30572997f34856251e3d66f)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[hub/powertoys/text-extractor.md](https://github.com/taras-janea/windows-dev-docs/blob/TextExtractor-20388/hub/powertoys/text-extractor.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/8f0bf4e1-986f-a415-e316-169a990c8a5d/202210031055073652-4065/BuildReport?accessString=ef8689cbf464d788f5118bb4fec51c0c96aa808f762c81a11b2304ff69ebec90).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/8f0bf4e1-986f-a415-e316-169a990c8a5d?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

Jay-o-Way commented on hub/powertoys/text-extractor.md:
```diff
@@ -14,9 +14,14 @@ Text Extractor is a convenient way to copy text from anywhere on your screen. Th
-By holding <kbd>shift</kbd>, you will change from adjusting the capture region's size to moving the capture region. When you release <kbd>shift</kbd>, you will be able to resize again.
+By holding <kbd>Shift</kbd>, you will change from adjusting the capture region's size to moving the capture region. When you release <kbd>Shift</kbd>, you will be able to resize again.
 
 ## Things to note
```
You could also place this in a special `Note` box, see https://learn.microsoft.com/contribute/markdown-reference#alerts-note-tip-important-caution-warning

taras-janea replied:
I'd prefer to leave it as it is.

Jay-o-Way reviewed (commented):
Line 28 _could_ be simplified: "1. This tool uses Optical Character Recognition to read the text on the screen."

taras-janea commented:
> Line 28 _could_ be simplified: "1. This tool uses Optical Character Recognition to read the text on the screen."

I'd prefer not to change it.

Court72 commented:
@mattwojo

Can you review the proposed changes? *IMPORTANT*: When the changes are ready for publication, add a `#sign-off` comment to signal that the PR is ready for the review team to merge. 

#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

crutkas commented on hub/powertoys/text-extractor.md:
````diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
+### How to query for OCR language packs
+
+The next command returns the list (PowerShell run as Administrator):
+```console
+PS C:\Windows\system32> Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*' }
+```
````
```suggestion
```powershell
Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*' }
```
```

crutkas commented on hub/powertoys/text-extractor.md:
````diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
+Text Extractor can only recognize languages that have the OCR language pack installed.
+
+The list can be obtained via PowerShell by running the following commands:
+```console
+PS C:\Windows\system32> [Windows.Media.Ocr.OcrEngine, Windows.Foundation, ContentType = WindowsRuntime]
+PS C:\Windows\system32> [Windows.Media.Ocr.OcrEngine]::AvailableRecognizerLanguages
+```
````
```suggestion
```powershell
[Windows.Media.Ocr.OcrEngine, Windows.Foundation, ContentType = WindowsRuntime]
[Windows.Media.Ocr.OcrEngine]::AvailableRecognizerLanguages
```
```

crutkas replied:
My change was larger. Having the c: blah blah blocks a copy paste

taras-janea replied:
Sorry, missed that part. Removed the PS C:\Windows\system32> prefix as well.

crutkas commented on hub/powertoys/text-extractor.md:
````diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
+```console
+PS C:\Windows\system32> $Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
+PS C:\Windows\system32> $Capability | Add-WindowsCapability -Online
+```
````
```suggestion
```powershell
$Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
$Capability | Add-WindowsCapability -Online
```
```

crutkas commented on hub/powertoys/text-extractor.md:
````diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
+### How to remove an OCR language pack
+
+The following commands remove the OCR pack for "en-US":
+```console
+PS C:\Windows\system32> $Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
+PS C:\Windows\system32> $Capability | Remove-WindowsCapability -Online
+```
````
```suggestion
```powershell
$Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
$Capability | Remove-WindowsCapability -Online
```
```

crutkas commented on hub/powertoys/text-extractor.md:
```diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
+
+## Troubleshooting
+
+- "No Possible OCR languages are installed." message
```
```suggestion
### "No Possible OCR languages are installed." message
```

crutkas reviewed (commented):
tweak to my suggestion and approved

taras-janea committed:
TextExtractor-20388: use powershell instead of console syntax highlight

opbld31 commented:
Learn Build status updates of commit _[21d5fc2](https://github.com/taras-janea/windows-dev-docs/commits/21d5fc2df288ba86fd1cf800951efaaa697a10b0)_: 

### :x: Validation status: errors

Please follow instructions [here](https://review.learn.microsoft.com/en-us/help/onboard/faq?branch=main#build--publish) which may help to resolve issue.


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-7b70d2eb)

<a id="7b70d2eb"></a>
###
 - **[Error: CannotMergeCommit]** `````Cannot merge commit 21d5fc2df288ba86fd1cf800951efaaa697a10b0 in branch TextExtractor-20388 of repository https://github.com/taras-janea/windows-dev-docs into branch docs (commit 8dc5a0eb2e2bfae1b820836647a70937e37e9bc6). Please follow this documentation: https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/ to use git.exe to resolve you content conflicts locally and then push to remote.`````

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/8f0bf4e1-986f-a415-e316-169a990c8a5d/202210131412047386-4065/BuildReport?accessString=b4f436f9a35732d789c4d6169929f90b2112940e07a9b6f6f6e89f0341ed61c7).

If you see build warnings/errors with permission issues, it might be due to single sign-on (SSO) enabled on Microsoft's GitHub organizations. Please follow instructions [here](https://review.learn.microsoft.com/en-us/help/onboard/sso-troubleshooting?branch=main) to re-authorize your GitHub account to Learn Build.

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/8f0bf4e1-986f-a415-e316-169a990c8a5d?tabName=brokenlinks).

**Note:** Your PR may contain errors or warnings or suggestions unrelated to the files you changed. This happens when external dependencies like GitHub alias, Microsoft alias, cross repo links are updated. Please use these [instructions](https://review.learn.microsoft.com/en-us/help/onboard/faq?branch=main&tabs=pr-validation-error-solution#q-why-do-i-get-pr-validation-result-with-cryptic-error-or-warning-messages-not-related-to-the-changes-i-make-in-my-working-branch) to resolve them.

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

opbld32 commented:
Learn Build status updates of commit _[5002d6b](https://github.com/taras-janea/windows-dev-docs/commits/5002d6ba89e3694a5e7bf04e2a7c59f47904ebfe)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[hub/powertoys/text-extractor.md](https://github.com/taras-janea/windows-dev-docs/blob/TextExtractor-20388/hub/powertoys/text-extractor.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/8f0bf4e1-986f-a415-e316-169a990c8a5d/202210131424399349-4065/BuildReport?accessString=da3383e31dab9bbd911d11f1ea9d96f51b8600c080f4b8f19da071f57c870662).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/8f0bf4e1-986f-a415-e316-169a990c8a5d?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

crutkas commented on hub/powertoys/text-extractor.md:
````diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
+
+The list can be obtained via PowerShell by running the following commands:
+```powershell
+PS C:\Windows\system32> [Windows.Media.Ocr.OcrEngine, Windows.Foundation, ContentType = WindowsRuntime]
````
The c: stuff needs to be removed. My suggestion had this removal included.

taras-janea replied:
Sorry, missed that part. Removed the PS C:\Windows\system32> prefix as well.

crutkas commented on hub/powertoys/text-extractor.md:
````diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
+
+The next command returns the list (PowerShell run as Administrator):
+```powershell
+PS C:\Windows\system32> Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*' }
````
The c: stuff needs to be removed. My suggestion had this removal included.

taras-janea replied:
Sorry, missed that part. Removed the PS C:\Windows\system32> prefix as well.

crutkas commented on hub/powertoys/text-extractor.md:
````diff
@@ -31,3 +36,65 @@ From the Settings menu, the following options can be configured:
+
+The following commands install the OCR pack for "en-US":
+```powershell
+PS C:\Windows\system32> $Capability = Get-WindowsCapability -Online | Where-Object { $_.Name -Like 'Language.OCR*en-US*' }
````
The c: stuff needs to be removed. My suggestion had this removal included.

taras-janea replied:
Sorry, missed that part. Removed the PS C:\Windows\system32> prefix as well.

taras-janea committed:
TextExtractor-20388: remove commands prefix "PS C:\Windows\system32>" according to comments

opbld33 commented:
Learn Build status updates of commit _[d850dfb](https://github.com/taras-janea/windows-dev-docs/commits/d850dfbeee2a557a38d77b5c4fe7dfcb718c334c)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[hub/powertoys/text-extractor.md](https://github.com/taras-janea/windows-dev-docs/blob/TextExtractor-20388/hub/powertoys/text-extractor.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/8f0bf4e1-986f-a415-e316-169a990c8a5d/202210131708488605-4065/BuildReport?accessString=995770da0664cfbf3e55da1f0b54c898d4b7a1ee2bbe0cb325b99ae372ce6894).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/8f0bf4e1-986f-a415-e316-169a990c8a5d?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

crutkas commented:
@mattwojo should be good pending your sign off

mattwojo committed:
Minor review updates

- Updated into to include a bit about the ability to grab text from images or videos, which seems like a central value prop. Also want to avoid words like "convenient" as that is very subjective.
- Formatting fixes for code blocks (don't combine separate commands into one block, only use supported language tags (unfortunately "console" isn't officially supported), needed some spacing
- Added brief lang-loc explainer

Looks great!

opbld32 commented:
Learn Build status updates of commit _[ae03dfb](https://github.com/taras-janea/windows-dev-docs/commits/ae03dfbbf4b753b2530ef0eea85e2fab15dff2bf)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[hub/powertoys/text-extractor.md](https://github.com/taras-janea/windows-dev-docs/blob/TextExtractor-20388/hub/powertoys/text-extractor.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/8f0bf4e1-986f-a415-e316-169a990c8a5d/202210131944326954-4065/BuildReport?accessString=2f886f154e589d99904b7834fe4528a4862ad24f8ad644c38ea66ca46d473e5c).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/8f0bf4e1-986f-a415-e316-169a990c8a5d?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

mattwojo commented:
@crutkas - I tweaked the intro statement a bit to add a bit about grabbing text from an image or video (to provide more context for most common use-cases) and removed the "make it more convenient" statement as we try to avoid these sort of subjective statements like "it's easy". 

At some point it may be worth also doing an accessibility review of the PowerToys docs... there are a lot of instances of instruction for using a mouse and right-clicking (for example), this may be unavoidable, but we are generally trying to write for folks who may be using assistive technology as well (like alternatives to a mouse -- like a trackpad or... IDK!) ...just a thought for future discussion.

If that intro looks ok, please just add #sign-off to get this merged live. :)
